src/vacation_calendar/chore.py

```diff
@@ -0,0 +1,185 @@
+from os.path import exists
+from enum import StrEnum
+import json
+import pandas as pd
+import numpy as np
+import datetime
+import holidays
+from typing import Union
+from functools import singledispatch
+
+# NOTE: the following lines will be taken from settings
+CONFIG_PATH = "config.json"
+current_year = datetime.datetime.now().year
+country = "IT"
+subdiv = "BO"
+WORKING_HOURS = 8
+MONTHLY_ROL = 8.66
+MONTHLY_VAC = 13.34
+AP_ROL = 20.3  # from config (ROL anno precedente)
+AP_VAC = 25.5  # from config) (VAC anno precedente)
+
+
+class TimeOffType(StrEnum):
+    VAC = "VACATION"
+    ROL = "PTO"  # Paid Time Off
+    VAC_MAT = "VAC_MAT"
+    ROL_MAT = "PTO_MAT"
+    VAC_RES = "VAC_RES"
+    ROL_RES = "PTO_RES"
+
+
+def read_config():
+    # NOTE: this will be probably removed using pydantic
+    if not exists(CONFIG_PATH):
+        d = {}
+    else:
+        with open(CONFIG_PATH, "r") as f:
+            d = json.load(f)
+    return d
+
+
+def create_base_calendar() -> pd.DataFrame:
+    """Generate the base calendar, with holidays and weekends corresponding to the config files
+    Returns:
+        (pd.DataFrame): the calendar dataframe
+    """
+    year_holidays = holidays.country_holidays(country, years=current_year, subdiv=subdiv)
+    start = datetime.datetime(current_year, month=1, day=1)
+    end = datetime.datetime(current_year, month=12, day=31)
+    dates = pd.date_range(start=start, end=end, freq="D")
+    dates_df = dates.to_frame(index=False, name="DATE")
+    dates_df["MONTH"] = dates_df.DATE.apply(lambda ser: ser.month)
+    dates_df["YEAR"] = dates_df.DATE.apply(lambda ser: ser.year)
+    dates_df["IS_WEEKEND"] = dates_df["DATE"].dt.dayofweek.isin([5, 6])
+    dates_df["HOLIDAY"] = (
+        dates_df["DATE"]
+        .apply(
+            lambda day: year_holidays[day] if day in year_holidays else pd.NA,
+        )
+        .astype("category")
+    )
+    dates_df["IS_HOLIDAY"] = np.where(dates_df["HOLIDAY"].notna(), True, False)
+    dates_df[TimeOffType.VAC] = 0.0
+    dates_df[TimeOffType.ROL] = 0.0
+    return dates_df
+
+
+@singledispatch
+def add_to_calendar(
+    date: Union[datetime.date, str], df: pd.DataFrame, value: float, ttype: TimeOffType
```

Comment:
The `Union` import can be avoided in favor of the new | simpler syntax, just for style.
```suggestion
    date: datetime.date | str, df: pd.DataFrame, value: float, ttype: TimeOffType
```